app/controllers/courses_controller.rb

```diff
@@ -726,34 +730,44 @@ def detect_and_convert_roster(roster)
+      # Email(9 - skip), College(10), Department(11 - skip), Major(12),
+      # Class(13), Graduation Semester(14 - skip), Units(15 - skip), Grade Option(16)
+      # ... the remaining fields are all skipped but shown for completeness
+      # QPA Scale(17), Mid-Semester Grade(18), Primary Advisor(19), Final Grade(20),
+      # Default Grade(21), Time Zone Code(22), Time Zone Description(23), Added By(24),
+      # Added On(25), Confirmed(26), Waitlist Position(27), Units Carried/Max Units(28),
+      # Waitlisted By(29), Waitlisted On(30), Dropped By(31), Dropped On(32), Roster As Of Date(33)
+      map = [-1, 8, 5, 6, 10, 12, 13, 16, -1, 3, 2]
       select_columns = ROSTER_COLUMNS_F20
-    when ROSTER_COLUMNS_F16
+    when ROSTER_COLUMNS_F16 # 32 fields
       # In CMU S3 roster. Columns are:
-      # Semester(0), Course(1), Section(2), (Lecture-skip)(3), (Mini-skip)(4),
-      # Last Name(5), First Name(6), (MI-skip)(7), Andrew ID(8),
-      # (Email-skip)(9), School(10), (Department-skip)(11), Major(12),
-      # Year(13), (skip)(14), Grade Policy(15), ...
-      map = [0, 8, 5, 6, 10, 12, 13, 15, -1, 1, 2]
+      # Semester(0 - skip), Course(1 - skip), Section(2), Lecture(3), Mini(4 - skip),
+      # Last Name(5), Preferred/First Name(6), MI(7 - skip), Andrew ID(8),
+      # Email(9 - skip), College(10), Department(11), Major(12),
+      # Class(13), Graduation Semester(14 - skip), Units(15 - skip), Grade Option(16)
+      # ... the remaining fields are all skipped but shown for completeness
+      # QPA Scale(17), Mid-Semester Grade(18), Primary Advisor(19), Final Grade(20),
+      # Default Grade(21), Added By(22), Added On(23), Confirmed(24), Waitlist Position(25),
+      # Units Carried/Max Units(26), Waitlisted By(27), Waitlisted On(28), Dropped By(29),
+      # Dropped On(30), Roster As Of Date(31)
+      map = [-1, 8, 5, 6, 10, 12, 13, 16, -1, 3, 2]
       select_columns = ROSTER_COLUMNS_F16
-    when ROSTER_COLUMNS_S15
+    when ROSTER_COLUMNS_S15 # 29 fields
       # In CMU S3 roster. Columns are:
-      # Semester(0), Lecture(1), Section(2), (skip)(3), (skip)(4), Last Name(5),
+      # Semester(0 - skip), Lecture(1), Section(2), (skip)(3), (skip)(4), Last Name(5),
       # First Name(6), (skip)(7), Andrew ID(8), (skip)(9), School(10),
-      # Major(11), Year(12), (skip)(13), Grade Policy(14), ...
-      map = [0, 8, 5, 6, 10, 11, 12, 14, -1, 1, 2]
+      # Major(11), Year(12), (skip)(13), Grade Policy(14), ... [elided]
+      map = [-1, 8, 5, 6, 10, 11, 12, 14, -1, 1, 2]
       select_columns = ROSTER_COLUMNS_S15
     else
       # No header row. Columns are:
-      # Semester(0), Email(1), Last Name(2), First Name(3), School(4),
-      # Major(5), Year(6), Grade Policy(7), (skip)(8), Lecture(9),
-      # Section(10), ...
+      # Semester(0 - skip), Email(1), Last Name(2), First Name(3), School(4),
+      # Major(5), Year(6), Grade Policy(7), Course(8 - skip), Lecture(9),
+      # Section(10)
       return parsedRoster
```

Comment:
To be fully consistent with what we're doing with SIO rosters (i.e. dropping the Semester and Course columns here, even though they will be ignored later on anyway), we should technically use a "map" here `[-1, 1, 2, 3, 4, 5, 6, 7, -1, 9, 10]`. However, this doesn't make a difference unless for some reason we make use of those columns in the future.